python/Ganga/Core/InternalServices/Coordinator.py

```diff
@@ -234,10 +180,10 @@ def enableInternalServices():
     Repository_runtime.bootstrap()
 
     # make sure all required credentials are valid
-    missing_cred = getMissingCredentials()
-    if missing_cred:
-        log.error("The following credentials are still required: %s."
-                  "Make sure you renew them before reactivating this session" % ','.join(missing_cred))
+    invalid_afs = [afsToken for afsToken in credential_store.get_all_matching_type(AfsToken) if not afsToken.is_valid()]
+
+    if invalid_afs:
+        log.error('No valid AFS token was found. Please re-authorise before reactivating this session.')
```

Comment:
I've tried to avoid changing the logic behind the runtime systems too much. For now I just wanted to maintain the current behaviour. I agree in general though that having a startup check for an invalid AFS token may be a good idea. How does it react at present?